pkg/providers/vsphere/vsphere_test.go

```diff
@@ -2007,89 +2035,62 @@ func TestSetupAndValidateCreateClusterTemplateDifferent(t *testing.T) {
 }
 
 func TestSetupAndValidateCreateClusterTemplateDoesNotExist(t *testing.T) {
-	ctx := context.Background()
-	clusterSpec := givenEmptyClusterSpec()
-	fillClusterSpecWithClusterConfig(clusterSpec, givenClusterConfig(t, testClusterConfigMainFilename))
-	provider := givenProvider(t)
-	govc := givenGovcMock(t)
-	provider.providerGovcClient = govc
-	setupContext(t)
+	tt := newProviderTest(t)
 
-	govc.EXPECT().ValidateVCenterSetup(ctx, provider.datacenterConfig, &provider.selfSigned).Return(nil)
-	govc.EXPECT().ValidateVCenterSetupMachineConfig(ctx, provider.datacenterConfig, provider.machineConfigs[clusterSpec.Spec.ControlPlaneConfiguration.MachineGroupRef.Name], &provider.selfSigned).Return(nil)
-	govc.EXPECT().ValidateVCenterSetupMachineConfig(ctx, provider.datacenterConfig, provider.machineConfigs[clusterSpec.Spec.WorkerNodeGroupConfigurations[0].MachineGroupRef.Name], &provider.selfSigned).Return(nil)
-	govc.EXPECT().ValidateVCenterSetupMachineConfig(ctx, provider.datacenterConfig, provider.machineConfigs[clusterSpec.Spec.ExternalEtcdConfiguration.MachineGroupRef.Name], &provider.selfSigned).Return(nil)
-	govc.EXPECT().SearchTemplate(ctx, provider.datacenterConfig.Spec.Datacenter, provider.machineConfigs[clusterSpec.Spec.ControlPlaneConfiguration.MachineGroupRef.Name]).Return("", nil)
+	tt.setExpectationsForDefaultDiskGovcCalls()
+	tt.setExpectationForVCenterValidation()
+	tt.setExpectationsForMachineConfigsVCenterValidation()
```

Comment:
these are the same expectations as before but dried out to reuse them from other tests